validators.rb

```diff
@@ -0,0 +1,57 @@
+class DynamicValidator
+  def initialize(attribute, rules)
+    @attribute = attribute
+    @rules = rules
+  end
+  def generate_validations(klass)
+    @rules.each do |rule|
+      define_validation_method(klass, rule)
+    end
+  end
+  private
+  def define_validation_method(klass, rule)
+    method_name = "validate_#{rule}_of_#{@attribute}"
+    attribute = @attribute.to_sym
+    klass.class_eval do
+      define_method(method_name) do
+        value = send(attribute)
+        case rule
+        when :presence
+          if value.nil? || (value.respond_to?(:empty?) && value.empty?)
+            puts "#{attribute.capitalize} can't be empty"
+          end
+        when :length
+          if value.respond_to?(:length) && value.length < 5
+            puts "#{attribute.capitalize} length should be at least 5 characters"
+          end
+        when :age_limit
+          if attribute == :age && (value.nil? || value < 18)
```

Comment:
.presence evaluates first right as per your input.
You do not need to check .nil? here again